### PR TITLE
Changing underpacing formula to consider duration less than one day

### LINKED ISSUE #57: Calculate difference at hours level if it is shorter than a day.

### PR DESCRIPTION
**Pull request by hanj7221**
Closes #57

### CODE REVIEW

File: lib/src/excel.dart [hunk 1]
```diff
@@ -216,17 +218,24 @@ class ExcelDart {
...
       // (spent / budget) / (current duration / flight duration)
       //
-      // Flight duration is calculated by (end date - start date + 1), and
-      // current duration is calculated by (now - start date + 1), so it is
-      // guaranteed that divide by zero will not happen.
-      /// TODO: calculate difference at hours level if it is shorter than a day.
-      /// Issue: https://github.com/googleinterns/dv360-excel-plugin/issues/57
+      // If current duration or flight duration is less than 1 day, then
+      // the formula calculates:
+      // (spent / budget) / (time elapsed since flight start / 24)
```
Comment (arulprakash-g) [inline lib/src/excel.dart]: in hours

File: lib/src/excel.dart [hunk 2]
```diff
@@ -216,17 +218,24 @@ class ExcelDart {
...
+      // Time elapsed has the granularity of hours and minutes.
+      // i.e. TEXT("1/1/2020 10:20:00 AM" - "1/1/2020 12:00:00 AM", "h.mm")
+      //      = 10.2
       final formula = ''' 
-       ($_spentColumn/$_budgetColumn) / 
-       (
-         (DATEDIF($_startDateColumn, NOW(), "D") + 1) /
-         (DATEDIF($_startDateColumn, $_endDateColumn, "D") + 1)
-       ) < $_thresholdAddress
+      IF(OR(DATEDIF($_startDateColumn, $_endDateColumn, "D") = 0,
```
Comment (arulprakash-g) [inline lib/src/excel.dart]: Consider splitting this into 2. First compute pacing % & then compare against threshold. Do we expose pacing % in the sheet?

File: lib/src/excel.dart [hunk 3]
```diff
@@ -216,17 +218,24 @@ class ExcelDart {
...
-       ) < $_thresholdAddress
+      IF(OR(DATEDIF($_startDateColumn, $_endDateColumn, "D") = 0, 
+            DATEDIF($_startDateColumn, NOW(), "D") = 0),
+         ($_spentColumn/$_budgetColumn) / 
+         (
+           TEXT(NOW()-$_startDateColumn, "h.mm") / 24
+         ), 
+         ($_spentColumn/$_budgetColumn) /
+         (
+           DATEDIF($_startDateColumn, NOW(), "D") /
```
Comment (arulprakash-g) [inline lib/src/excel.dart]: Is this only for active IOs? If not (now - startDate) can be -ve for segments in future.

**Review by arulprakash-g — APPROVED** [review]

File: lib/src/excel.dart (hunk 3 above)
Comment (hanj7221) [reply lib/src/excel.dart]: Yes, we are only querying for IOs that have entity status of active. And for all the active IOs, i will also filter out those that are not in flight.

**Review by ruben-aghayan — APPROVED** [review]

**Pull request merged**